ui/js/graphs.js

```diff
@@ -764,11 +780,73 @@ perf.controller('TestChooserCtrl', function($scope, $modalInstance, $http,
         $modalInstance.dismiss('cancel');
     };
 
+    $scope.testsToAdd = []; // tests in the "tests to add" list
     $scope.unselectedTestList = []; // tests in the "tests" list
     $scope.selectedTestSignatures = []; // tests in the "tests" list that have been selected by the user
-    $scope.testsToAdd = []; // tests in the "tests to add" list
     $scope.selectedTestsToAdd = []; // tests in the "to add" test list that have been selected by the user
 
+    var relateSeries = options.relatedSeries;
+    // if we choose to add all related platforms, firstly we need
+    // to obtain all the platform of this project, then pick out the
+    // specific test by platform and project.
+    var addRelatePlaform = function(relateSeries) {
+        PhSeries.getAllSeries(relateSeries.projectName,
+            $scope.timeRange, optionCollectionMap).then(
+            function(seriesData) {
+                var platformList = seriesData.platformList;
+                platformList.forEach(function(platform) {
+                    var testList = _.sortBy(_.filter(seriesData.seriesList,
+                        { platform: platform }), 'name');
+                    var temp = _.findWhere(testList, {"name": relateSeries.name});
+                    // if found something different from the series we already have,
+                    // then we push it into the testsToAdd list.
+                    if (temp !== undefined && temp.signature !== relateSeries.signature) {
+                        $scope.testsToAdd.push(_.clone(temp));
+                    }
+                });
+            }
+        );
+        return $scope.testsToAdd.length;
```

Comment:
it will always return undefined because PhSeries.getAllSeries(.....) haven't finish yet.